thinc/backends/ops.py

```diff
@@ -252,7 +284,7 @@ def flatten(
             result = xp.asarray(result, dtype=dtype)
         return result
 
-    def unflatten(self, X: Floats2d, lengths: Ints1d, pad: int = 0) -> List[Floats2d]:
+    def unflatten(self, X: ArrayTXd, lengths: Ints1d, pad: int = 0) -> List[ArrayTXd]:
```

Comment:
Definitely good to widen the scope of such functions - I think these were too strictly typed.